request.go

```diff
@@ -123,15 +128,29 @@ func (r *Request) execute(ctx context.Context, c Client, output interface{}) err
 			)
 			return false, errorMessage
 		}
-		c.cfg.logger.Debugf("Response body: %v", string(iostream))
-		//if output is set
-		if output != nil {
-			err = json.Unmarshal(iostream, output) //Edit the given struct
-			if err != nil {
-				c.cfg.logger.Errorf("Error while marshaling JSON: %v", err)
-				return false, err
-			}
-		}
+		logger.Debugf("Response body: %v", string(responseBodyBytes))
+
 		return false, nil
-	}, c.cfg.maxNumberOfRetries, c.cfg.delayInterval)
+	}, c.MaxNumberOfRetries(), c.DelayInterval())
+	//if retry fails
+	if err != nil {
+		return err
+	}
+
+	//if output is set
+	if output != nil {
+		//Unmarshal body bytes to the given struct
+		err = json.Unmarshal(responseBodyBytes, output)
+		if err != nil {
+			logger.Errorf("Error while marshaling JSON: %v", err)
+			return err
+		}
+	}
+
+	//If the client is synchronous, and the request does not skip
+	//checking a request, wait until the request completes
+	if c.Synchronous() && !r.skipCheckingRequest {
```

Comment:
I really don't get this line… and I don't understand the purpose of this `skipCheckingRequest` field. (Maybe you need to help me here.)

I read this as "if the request is blocking and nobody cares for checking the current request's status, continue waiting for completion". Is this right?

Since request type is now private and not exposed anymore, can't we just remove this ominous field "skipCheckingRequest"? It seems to be always true anyway.

Please correct me if I'm wrong.

Reply:
You can see that `waitForRequestCompleted` method also invokes `execute` function. Every request sent to gridscale server has its own request UUID. If we don't check `skipCheckingRequest`, the `waitForRequestCompleted` method keep calling itself forever 🗡 . Btw, all requests made by `GetXXX` methods are not check-able by their UUIDs (I have tried this case). On the other hand, Create/Update/Delete requests can be checked by sending their UUIDs to request endpoint.

Reply:
Cool. Thanks or the explanation  :+1: